tests/spec/fs.writeFile-readFile.spec.js

```diff
@@ -99,3 +99,118 @@ describe('fs.writeFile, fs.readFile', function() {
+    var fs = util.fs();
+
+    
+    fs.promises.readFile('/no-such-file', 'utf8')
+      .then(data => expect(data).not.to.exist)
+      .catch(error => { expect(error).to.exist; expect(error.code).to.equal('ENOENT'); done(); });
+    
+
+  });
+
+  it('should write, read a utf8 file without specifying utf8 in writeFile', function(done) {
+    var fs = util.fs();
+    var contents = 'This is a file.';
+
+    fs.promises.writeFile('/myfile', contents)
+      .then( () => {
+
+        fs.promises.readFile('/myfile', 'utf8')
+          .then( (data, error) => { expect(data).to.equal(contents); expect(error).not.to.exist; done(); });
+
+      })
+      .catch(error => { throw error; });
+  });
+
+  it('should write, read a utf8 file with "utf8" option to writeFile', function(done) {
+    var fs = util.fs();
+    var contents = 'This is a file.';
+
+    fs.promises.writeFile('/myfile', contents, 'utf8') 
+      .then( () => {
+
+        fs.promises.readFile('/myfile', 'utf8')
+          .then( (data, error) => { expect(data).to.equal(contents); expect(error).not.to.exist; done(); });
+
+      })  
+      .catch(error => { throw error; });
+  });
+
+  it('should write, read a utf8 file with {encoding: "utf8"} option to writeFile', function(done) {
+    var fs = util.fs();
+    var contents = 'This is a file.';
+
+    fs.promises.writeFile('/myfile', contents, { encoding: 'utf8' })
+      .then( () => {
+
+        fs.promises.readFile('/myfile', 'utf8')
+          .then( (data, error) => { expect(data).to.equal(contents); expect(error).not.to.exist; done(); });
+
+      }) 
+      .catch(error => { throw error; });
+  });
+
+  it('should write, read a binary file', function(done) {
+    var fs = util.fs();
+    // String and utf8 binary encoded versions of the same thing: 'This is a file.'
+    var binary = new Buffer([84, 104, 105, 115, 32, 105, 115, 32, 97, 32, 102, 105, 108, 101, 46]);
+
+    fs.promises.writeFile('/myfile', binary)
+      .then( () => {
+
+        fs.promises.readFile('/myfile')
+          .then( (data, error) => { expect(data).to.deep.equal(binary); expect(error).not.to.exist; done(); });
+
+      }) 
+      .catch(error => { throw error; });
+  });
+
+  it('should follow symbolic links', function(done) {
+    var fs = util.fs();
+    var contents = 'This is a file.';
+
+    fs.promises.writeFile('/myfile', '', { encoding: 'utf8' })
+      .then( () => {
+        fs.promises.symlink('/myfile', '/myFileLink')
+          .then( () => {
+            fs.promises.writeFile('/myFileLink', contents, 'utf8')
+              .then( () => {
+
+                fs.promises.readFile('/myFileLink', 'utf8')
+                  .then( (data, error) => { expect(data).to.equal(contents); expect(error).not.to.exist; done(); });
+                  
+              }) 
+              .catch(error => { throw error; });   
+          }) 
+          .catch(error => { throw error; });
+      }) 
+      .catch(error => { throw error; });
```

Comment:
Instead of having 3 of these, I think we can just have one final `catch` on the outer promise chain.